csrc/scheduler/mma_utils.cpp

```diff
@@ -774,56 +774,58 @@ void WarpMmaSwizzler::scheduleOperandRead(TensorView* tv, MmaOptions options) {
 void WarpMmaSwizzler::scheduleMmaWarpOutput(
     TensorView* tv,
     MmaOptions options) {
-  // Assume last 2 dims [M16, N8] or [M16, N8, R]
-  // Locate instruction m
-  bool is_reduction = tv->axis(-1)->isReduction();
-
-  int m_pos = is_reduction ? -3 : -2;
-
-  //  m
-  // [16, 8  (,R)]
-  tv->split(m_pos, 8);
-  tv->split(m_pos + 1, 8);
-  //        m
-  // [2, 8, 1, 8  (,R)]
-  tv->split(m_pos + 1, 2);
-
-  //              m
-  // [2o, 8o, 1, 4i, 2i (,R)]
-  tv->reorder({{m_pos, m_pos - 1}, {m_pos - 1, m_pos}});
-  m_pos--;
-  //           m
-  // [2o, 8o, 4i, 1, 2i (,R)]
-  tv->merge(m_pos - 1);
+  // This function works for all mma ops, regardless of the architecture. The
+  // Hopper one is the most general one. For earlier architectures, we will have
+  // some dimensions with size 1 after split, this is fine.
+  // Memory format for hopper mma:
+  // https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#wgmma-64n16-d
+
+  // Assume last 2 dims, for example [M64, N24] or [M64, N24, R]
+  NVF_ERROR(tv->nDims() >= 2);
+  bool is_mma_output = tv->definition()->isA<MmaOp>();
```

Comment:
The `applyMmaSwizzle` is also used to manually schedule some tests. For example, if you have
```C++
T1 = mma(...)
T2 = relu(T1)
```
Then you can also do `T2->applyMmaSwizzle` to schedule `T2`, instead of schedule `T1` and propagate. In general, I think we should prefer the schedule one and propagate to others approach, but that needs some extra work to clean up the tests.

Reply:
Ok, it makes sense although it also means that we have changed production code to facilitate tests. Not perfect but I understand the reason.

I agree with you that the approach with schedule-one-and-propagate-to-others should be our main target.

Thank you for sharing explanation!